helix-view/src/input.rs

```diff
@@ -387,6 +387,21 @@ impl std::str::FromStr for KeyEvent {
                     .then_some(KeyCode::F(function))
                     .ok_or_else(|| anyhow!("Invalid function key '{}'", function))?
             }
+            _ if s.ends_with('-')
+                && tokens.last().map(|s| str::is_empty(s)).unwrap_or_default() =>
+            {
+                if tokens.len() > 1 {
```

Comment:
I think the tokens stuff is a little hard to follow. We can check for the "is `-` and doesn't have modifiers" case easier with `s == "-"` and then use `tokens.clear()` and return `KeyCode::Char('-')` in that branch.

I like the explicit error for this case recommending using `keys::MINUS`. We can expand on it a little by recommending the full key sequence:

```rust
let mut suggestion = s[..s.len() - 1].to_string();
suggestion.push_str(keys::MINUS);
```

and suggesting that in the error message

Reply:
good point - this can be a lot simpler now